pkg/controller-gen/main.go

```diff
@@ -265,6 +276,54 @@ func generateClientset(groups map[string]bool, customArgs *cgargs.CustomArgs) er
 	)
 }
 
+func generateOpenAPI(groups map[string]bool, customArgs *cgargs.CustomArgs) error {
+	if len(groups) == 0 {
+		return nil
+	}
+
+	openAPIArgs := oaargs.New()
+	openAPIArgs.OutputDir = filepath.Join(customArgs.OutputBase, customArgs.Options.OutputPackage, "openapi")
+	openAPIArgs.OutputFile = "zz_generated_openapi.go"
+	openAPIArgs.OutputPkg = customArgs.Options.OutputPackage + "/openapi"
+	openAPIArgs.GoHeaderFile = customArgs.Options.Boilerplate
+
+	if err := openAPIArgs.Validate(); err != nil {
+		return err
+	}
+
+	getTargets := func(context *generator.Context) []generator.Target {
+		return oa.GetTargets(context, openAPIArgs)
+	}
```

Comment:
Nit: I'd move this variable closer to where it's used below.

Reply:
Thanks!